src/connectors/impls/http/auth.rs

```diff
@@ -12,14 +12,30 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use crate::errors::Result;
+use elasticsearch::auth::{ClientCertificate, Credentials};
+
+use crate::{
+    connectors::utils::tls::{load_certs, load_keys},
+    errors::Result,
+};
 
 /// Authorization methods
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
 #[serde(rename_all = "lowercase")]
 pub enum Auth {
     #[serde(alias = "basic")]
     Basic { username: String, password: String },
+    #[serde(alias = "bearer")]
+    Bearer(String),
+    #[serde(alias = "client_certificate")]
+    ClientCertificate {
```

Comment:
Would make sense to move this into `utils::tls` and keep the same naming conventions we have in other TLS related configs?

Reply:
Why move it out of this enum? Or would you wrap this struct in a newtype and move the struct to tls?

Reply:
Ja I was thinking to create a struct fir client certificates, since they do exist for all TLS endpoints, not just HTTP, and wrap this here as a serde-flattened struct. Moving it to the `tls` utils will make it easier ti keep naming uniform through places, like not having `private_key` in one place and `key` in another

Reply:
Yeah makes sense. Will do.

Reply:
Done, `Auth` is no solely for populating the `Authorization` header.